pkg/deploy/service.go

```
@@ -62,7 +68,11 @@ func CreateDeploymentRequest(application string, config *model.OrchestrationConf
 				},
 			},
 			Constraints: &pipelineConstraint,
-		})
+		}
+		if analysis.DefaultAccount != nil {
```

I am guessing the reason you are making the decision of whether or not to add the analysis definition on defaultAccount is because the PRD states it is required. If that's the case it would be better to return an error here instead of making the request so that the user can be informed of their mistake. It looks to me like it will currently not add the analysis definition and then make the request to deploy engine, which will then fail. 

Actually, we should return that error before we start iterating the Targets. The PRD indicates that AccountType should required, however I think the comment about it being inferred applies and thus we don't need to check that.

---

Thinking about this more, we also probably only want to add the config to the deployment if that deployment has one or more analysis steps.

---

Ok, I addressed at least part of this, I think we need to have a discussion around how much validation we want to do in the CLI vs server side. Currently we do very little validation within the cli and rely heavily on the serverside to do it validation. 

If we want to expand the validation on the cli side we probably need to file a ticket to address that.